components/scim/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -1024,10 +1024,6 @@ public Group patchGroup(Group oldGroup, Group newGroup) throws CharonException {
                 List<String> addedMembers = new ArrayList<>();
                 List<String> deletedMembers = new ArrayList<>();
 
-                if (addRequestedMembers.isEmpty() && deleteRequestedMembers.isEmpty()) {
-                    String users[] = carbonUM.getUserListOfRole(newGroup.getDisplayName());
-                    deletedMembers = Arrays.asList(users);
-                }
 
```

Comment:
This doesn't seem to be the correct fix. According to SCIM spec if we send meta attribute with value 'members' for PATCH operations, all the members of the group must be removed. If you remove the above lines like above this won't happen.

I think the issue is in the way we are defining and evaluating the condition for deleting all the users if meta attribute 'members' is sent. We are indicating the removal of all the users by empty list. We expect the list to have some members if specifically members were specified to be deleted, or empty if meta 'members' was requested, OR NULL if nothing needs to happen. In this case the list never becomes NULL. The charon library seems to always return an empty list and that is why we delete all the members. Depending on the length of a returned collection to mean something is absolutely wrong IIHO. Please verify and if so let's try to fix that.